tool/tsh/tsh.go

```diff
@@ -3132,9 +3136,29 @@ func makeClientForProxy(cf *CLIConf, proxy string, useProfileLogin bool) (*clien
 	// addresses from the proxy (this is what Ping does).
 	if cf.IdentityFileIn != "" {
 		log.Debug("Pinging the proxy to fetch listening addresses for non-web ports.")
-		if _, err := tc.Ping(cf.Context); err != nil {
+		pr, err := tc.Ping(cf.Context)
+		if err != nil {
 			return nil, trace.Wrap(err)
 		}
+		tc.LoadAllCAs = pr.Auth.LoadAllHostCAs
```

Comment:
I think this should move into [applyProxySettings](https://github.com/gravitational/teleport/blob/f1b646eb35ad76648ae034472d29bc33da9adcea/lib/client/api.go#L3956) where we update teleport client with settings fetched from ping response.